### PR TITLE
enhance(logs): scrollable/windowed logs and actions sidebar

### PR DESCRIPTION
**Pull request by davidvader**
Proposing an enhancement to step logs to wrap them in scrollable/resizable window.

Depends on #193 

### UX Improvement

- Reduce overall screen size
- Improved scrolling performance
- More interactive/modern feel

### Performance

The following performance summaries were gathered and summarized over 20~ runs using Google Chrome's performance profiler, auto expand/collapsing 5 step logs each with 500+ lines.

##### Before

ui_expand_before

##### After

ui_expand_after

#### Conclusion

Reducing screen clutter and browser workload, time spent scripting/dynamically generating content is converted to idle time. However low sample size may be misleading. 

### Example

Screen Shot 2020-08-27 at 2 37 28 PM

### CODE REVIEW

**Comment by wass3r** [comment]
since we started customizing the scrollbar, what do you think about tweaking it to be closer to something like:

image

removing the left/right rails (1px vertical lines) and the treatment of the resize handle make the whole thing a bit more discreet and clean. thoughts?

**Comment by davidvader** [comment]
### Windowed Logs w/ Scrolling Actions
Screen Shot 2020-09-02 at 12 37 43 PM

File: src/scss/_main.scss [hunk 1]
```diff
@@ -879,7 +919,7 @@ nav {
 
 .step-error,
 .step-skipped {
-  margin-left: 1.5em;
+  padding: 1rem 1.5em;
```
Comment (davidvader) [inline src/scss/_main.scss:922]: moving the padding from logs-container to step-error

File: src/scss/_main.scss [hunk 2]
```diff
@@ -837,12 +823,54 @@ nav {
...
+  height: 22rem;
+  overflow-y: auto;
+
+  min-height: 10rem;
+  max-height: 75rem;
+  padding-bottom: 1rem;
+  padding-top: 1rem;
+  resize: vertical;
+
+  scrollbar-color: var(--color-coal-light) var(--color-bg-dark);
```
Comment (davidvader) [inline src/scss/_main.scss:852]: added to support firefox
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scrollbars

firefox also does not support restyling the automatic resizer :( 
https://caniuse.com/css-resize

File: src/scss/_main.scss [hunk 3]
```diff
@@ -1349,19 +1434,3 @@ nav {
 .overflow-auto {
   overflow: auto;
 }
-
```
Comment (davidvader) [inline src/scss/_main.scss:1352]: removing unused style

File: src/scss/_themes.scss [hunk 4]
```diff
@@ -105,6 +105,21 @@ body.theme-light {
...
+    border-bottom: 3px solid var(--color-bg-light);
+    border-right: 3px solid var(--color-bg-light);
+  }
+
+  .logs-table::-webkit-scrollbar-thumb {
+    background-color: var(--color-bg-light);
+    border-radius: 1rem;
+    border: 2px solid var(--color-bg);
+  }
+
```
Comment (davidvader) [inline src/scss/_themes.scss:122]: adding themed scrollbar style due to color-bg-dark matching with step background, hiding the scroll thumb

File: src/scss/_main.scss [hunk 5]
```diff
@@ -864,32 +892,87 @@ nav {
...
 
   color: var(--color-red-light);
   font-size: 14px;
 }
 
-.step-error .message,
-.step-skipped .message {
-  margin-left: 0.2em;
-}
-
```
Comment (davidvader) [inline src/scss/_main.scss:892]: this is handled with #199

File: src/scss/_main.scss [hunk 6]
```diff
@@ -864,32 +892,87 @@ nav {
...
+}
+
+.logs-sidebar .actions {
+  display: flex;
+  flex-direction: column;
+
+  button {
+    background: var(--color-coal-light);
+    margin-bottom: 0.4rem;
+    color: var(--color-offwhite);
```
Comment (davidvader) [inline src/scss/_main.scss:964]: style targeted for cross-theme compatibility

File: src/scss/_main.scss [hunk 7]
```diff
@@ -864,32 +892,87 @@ nav {
...
 
   color: var(--color-red-light);
   font-size: 14px;
 }
 
-.step-error .message,
-.step-skipped .message {
-  margin-left: 0.2em;
-}
-
```
Comment (davidvader) [inline src/scss/_main.scss:892]: solved in #202

File: src/elm/Pages/Build/View.elm [hunk 8]
```diff
@@ -612,15 +649,15 @@ stepError step =
...
+        [ span [ class "message" ] [ text "step was killed" ] ]
 
 
 {-| stepSkipped : renders message for a skipped step
 -}
 stepSkipped : Step -> Html msg
 stepSkipped _ =
     div [ class "step-skipped", Util.testAttribute "step-skipped" ]
-        [ span [ class "label" ] [ text "step was skipped" ] ]
+        [ span [ class "message" ] [ text "step was skipped" ] ]
```
Comment (davidvader) [inline src/elm/Pages/Build/View.elm:660]: solved in #202

File: src/scss/_main.scss [hunk 9]
```diff
@@ -864,32 +892,87 @@ nav {
...
+  height: 100%;
+  position: absolute;
+  right: 1rem;
+  padding-bottom: 2rem;
+}
+
+.logs-sidebar .inner-container {
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
```
Comment (wass3r) [inline src/scss/_main.scss:947]: i believe these can be safely removed

File: src/scss/_main.scss [hunk 10]
```diff
@@ -864,32 +892,87 @@ nav {
...
+.logs-header .actions {
+  padding: 0.3rem;
+
+  div {
+    justify-content: flex-end;
+  }
+
+  .button {
+    margin-right: 0.3rem;
+  }
```
Comment (wass3r) [inline src/scss/_main.scss:930]: should this just be a modifier on `.buttons`? looks like that is what this is targeting.

**Review by wass3r — COMMENTED** [review]
looking nice! well done.. couple of suggestions.

File: src/scss/_main.scss (hunk 10 above)
Comment (davidvader) [reply src/scss/_main.scss]: taken care of 👍 
https://github.com/go-vela/ui/pull/197/commits/0ebbf0d05a92a3deeec6ac94e30f4f35398e1ffd

File: src/scss/_main.scss (hunk 9 above)
Comment (davidvader) [reply src/scss/_main.scss]: taken care of 👍  also removed some other redundant style
https://github.com/go-vela/ui/pull/197/commits/0ebbf0d05a92a3deeec6ac94e30f4f35398e1ffd

**Review by wass3r — APPROVED** [review]

File: src/scss/_main.scss [hunk 11]
```diff
@@ -864,32 +883,81 @@ nav {
...
+  width: 100%;
+  z-index: 1;
+}
+
+.logs-header .actions {
+  padding: 0.3rem;
+
+  .buttons {
+    justify-content: flex-end;
+    margin-right: 0.3rem;
```
Comment (wass3r) [inline src/scss/_main.scss:917]: so, i actually meant adding a modifier class to https://github.com/go-vela/ui/blob/master/src/scss/_buttons.scss#L226-L229 and building that out. however, this is clearer. we can deal with that in a different PR.

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**